minard/burst.py

```diff
@@ -47,3 +45,93 @@ def burst_run_detail(run_number, subrun, sub):
         files = "%i_%i_%i" % (run_number,subrun,sub)
 
     return result, files
+
+def load_bursts_search(search, start, end, offset, limit):
+    """
+    Returns a dictionary with the burst runs loaded from couchdb.
+    The returned dictionary is given by one of the search conditions on the page:
+    either by run, date or GTID which all use corresponding couchdb views.
+    """
+    server = couchdb.Server("http://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst"]
+
+    results = []
+    skip = offset
+
+    if search == "run":
+
+        startkey = [int(start), 0, {}]
+        endkey = [int(end), {}]
+
+        view = '_design/burst/_view/burst_by_run'
+
+    elif search == "date":
+
+        start_year = start[0:4]
+        start_month = start[5:7]
+        start_day = start[8:10]
+        end_year = end[0:4]
+        end_month = end[5:7]
+        end_day = end[8:10]
+        if start_month[0] == "0":
+            start_month = start_month[1]
+        if end_month[0] == "0":
+            end_month = end_month[1]
+        if start_day[0] == "0":
+            start_day = start_day[1]
+        if end_day[0] == "0":
+            end_day = end_day[1]
+        startkey = [int(start_year), int(start_month), int(start_day)]
+        endkey = [int(end_year), int(end_month), int(end_day)]
+
+        view = '_design/burst/_view/burst_by_date'
+
+    elif search == "gtid":
+
+        view = '_design/burst/_view/burst_by_GTID'
+
+    if (search=="run") or (search=="date"):
```

Comment:
can delete the parentheses and add spaces around ==

Reply:
or `search in ('run','date')`

Reply:
done